book/src/getting-started/5-cmake-integration.md

```diff
@@ -0,0 +1,153 @@
+<!--
+SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# Building with CMake
+
+```diff,ignore
+- Disclaimer: The CMake integration for CXX-Qt is still WIP.
+- The current state is far from optimal and will likely improve
+- a lot in the future, so don't be discouraged by anything in
+- this chapter.
+- Contributions are also welcome.
+```
+
+Before we can get started on building Qt with CMake, we should first make our Cargo build ready for it.
+If you've generated your project with the `cargo new --lib` command, your Cargo.toml likely looks something like this:
+```toml,ignore
+[package]
+name = "cxx-qt-getting-started"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+```
+
+We'll have to do multiple things:
+- Instruct cargo to create a static lib with a defined name for CMake to link against ("rust").
+- Add `cxx`, `cxx-qt`, as well as `cxx-qt-lib` as dependencies.
+- Add `clang-format` and `cxx-qt-build` as build-dependencies.
+
+In the end, your `Cargo.toml` should look like this:
+```toml,ignore
+[package]
+name = "cxx-qt-getting-started"
+version = "0.1.0"
+edition = "2021"
+
+# Don't touch this, the CMake file expects a static lib with a specific name
+[lib]
+name = "rust"
+crate-type = ["staticlib"]
+
+[dependencies]
+cxx = "1.0"
+cxx-qt = "0.2"
+cxx-qt-lib = "0.2"
+
+[build-dependencies]
+clang-format = "0.1"
+cxx-qt-build = "0.2"
+```
+
+We'll then also need to add a script named `build.rs` next to our `Cargo.toml`:
+```rust,ignore
+{{#include ../../../examples/qml_minimal/build.rs:book_build_rs}}
+```
+This is what generates the C++ code for our `MyObject` class at compile-time.
```

Comment:
It's probably important to note that you need to specify all the files in which you specify the macro, otherwise it won't work. I've had it a few times where i've forgotten to add a new file to the list and wondered why things error oddly :-)

Reply:
Done